test/acceptance/features/steps/workercluster.py

```diff
@@ -185,3 +371,69 @@ def ensure_worker_cluster_running(context, cluster_name: str, primaza_cluster_na
     primaza_cluster = context.cluster_provider.get_primaza_cluster(primaza_cluster_name)
     p_csr_pem = primaza_cluster.create_certificate_signing_request_pem()
     worker_cluster.create_primaza_user(p_csr_pem)
+
+
+@given(u'On Worker Cluster "{cluster_name}", application namespace "{namespace}" exists')
+def ensure_application_namespace_exists(context, cluster_name: str, namespace: str):
+    worker = context.cluster_provider.get_worker_cluster(cluster_name)  # type: WorkerCluster
+    worker.create_application_namespace(namespace)
+
+
+@step(u'On Worker Cluster "{cluster_name}", Primaza Application Agent is deployed into namespace "{namespace}"')
+def application_agent_is_deployed(context, cluster_name: str, namespace: str):
+    worker = context.cluster_provider.get_worker_cluster(cluster_name)  # type: WorkerCluster
+
+    polling2.poll(
+        target=lambda: worker.is_app_agent_deployed(namespace),
+        step=5,
+        timeout=30)
+
+
+@step(u'On Worker Cluster "{cluster_name}", Primaza Application Agent is not deployed into namespace "{namespace}"')
+def application_agent_is_not_deployed(context, cluster_name: str, namespace: str):
+    worker = context.cluster_provider.get_worker_cluster(cluster_name)  # type: WorkerCluster
+
+    def is_not_found():
+        try:
+            worker.is_app_agent_deployed(namespace)
+        except ApiException as e:
+            return e.reason == "Not Found"
+        return False
+
+    polling2.poll(
+        target=lambda: is_not_found(),
+        step=5,
```

Comment:
Maybe reduce step sizes from 5 to 1?  This can _really_ slow down acceptance tests.

Reply:
that's a good idea, I'm going to apply the suggested change. We should also change this value in the other steps. I'd create a PR for that.